src/main/java/school/hei/haapi/service/UserService.java

```diff
@@ -178,6 +180,15 @@ public List<User> getByGroupId(String groupId) {
             groupedStudentsJoin.add(groupFlow.getStudent());
           }
         });
+
     return groupedStudentsJoin.stream().distinct().collect(toList());
   }
+
+  public List<User> getByGroupId(String groupId, PageFromOne page, BoundedPageSize pageSize) {
+    // Calculate start and end index for pagination
+    int startIndex = (page.getValue() - 1) * pageSize.getValue();
+    int endIndex = Math.min(startIndex + pageSize.getValue(), getByGroupId(groupId).size());
```

Comment:
store the getByGroupId return value in a variable first and use it for size and return x)